include/Dialect/Polynomial/IR/PolynomialAttributes.td

```diff
@@ -46,26 +47,29 @@ def Ring_Attr : Polynomial_Attr<"Ring", "ring"> {
       #ring = #polynomial.ring<cmod=1234, ideal=#polynomial.polynomial<x**1024 + 1>>
   }];
 
-  // The extra cwidth parameter is required because, when the MLIR framework's
-  // StorageUniquer is determining whether to build a new instance of this
-  // attribute or reuse an existing one, it does an == comparison on the
-  // parameters, and comparing two APInts with different underlying bit widths
-  // leads to an assertion failure. By adding the container bit width as a
-  // parameter, that bit width is compared first and the rest of the check is
-  // skipped if the bit widths disagree. The custom builder and custom
-  // parser/printer ensures that cwidth is automatically inferred from cmod, and
-  // not printed or parsed, so that the cwidth parameter need not be known to the
-  // user.
-  let parameters = (ins "int64_t": $cwidth, "APInt": $cmod, "Polynomial":$ideal);
+  let parameters = (ins "IntegerAttr": $cmod, "Polynomial":$ideal);
 
   let builders = [
-    AttrBuilderWithInferredContext<(ins "APInt": $cmod, "Polynomial":$ideal), [{
-      return $_get(ideal.getContext(), cmod.getBitWidth(), cmod, ideal);
+    AttrBuilderWithInferredContext<
```

Comment:
do you end up using this builder? i was slightly concerned about the use of `zextOrTrunc` here, so if it's unused (I don't see it in the diff, and it appears different) maybe best to remove it until tested or needed

Reply:
Oh, maybe this is used when the parser sees the `: i<width>` specified?

Reply:
Ah, yes when fixing my PR just now, I found out the default generated parser uses a builder (with IntegerAttr type param) like 
```
LWEParamsAttr::get(odsParser.getContext(),
      IntegerAttr((*_result_cmod)),
      unsigned((*_result_dimension)));
```

and it ensured that the `_result_cmod` was a parser that can gather the type attribute on the int.

> This is supposed to ensure that the integer type and the APInt's bit width always match.

I guess what I'm asking is - do you need it right now if you have no callers?

Reply:
Fair enough, I thought I was using it. Removed.